src/lib.rs

```diff
@@ -31,31 +56,42 @@ fn default_result<T>() -> Result<T, Error> {
 ///
 /// ```
 /// let saddr = "127.0.0.1:443".parse().unwrap();
-/// let mtu = mtu::interface_mtu(&saddr).unwrap();
-/// println!("MTU towards {saddr:?} is {mtu}");
+/// let mtu = mtu::interface_mtu((None, saddr)).unwrap();
+/// println!("MTU for {saddr:?} is {mtu}");
 /// ```
 ///
 /// # Errors
 ///
 /// This function returns an error if the local interface MTU cannot be determined.
-pub fn interface_mtu(remote: &SocketAddr) -> Result<usize, Error> {
+pub fn interface_mtu<A>(addrs: A) -> Result<usize, Error>
+where
+    SocketAddrs: From<A>,
+    A: std::marker::Copy + std::fmt::Debug,
+{
     #[cfg(any(target_os = "macos", target_os = "linux", target_os = "windows"))]
     #[allow(unused_assignments)] // Yes, res is reassigned in the platform-specific code.
     let mut res = default_result();
```

Comment:
There's a different way that I'd suggest you do this:

```rust
fn interface_mtu<...>(...) -> ... {
  // make the socket
  // ...
  let socket = UdpSocket::bind(...);
  // ...
  interface_mtu_impl(&socket)
}

#[cfg(linux/mac)]
fn interface_mtu_impl(socket: &UdpSocket) -> ... { 
  // linux/mac implementation
}

#[cfg(windows)]
fn interface_mtu_impl(socket: &UdpSocket) -> ... { 
  // windows implementation
}

#[cfg(not(linux/mac/windows))]
fn interface_mtu_impl(socket: &UdpSocket) -> ... { 
  Err(Error::whatever)
}
```

You might even leave off that last one so that this fails to compile on unsupported platforms.  I don't like that, but it's a valid option.